src/machine/machine_nrf52xxx.go

```diff
@@ -164,13 +169,7 @@ func (a ADC) Get() uint16 {
 	}
 	nrf.SAADC.EVENTS_STOPPED.Set(0)
 
-	value := int16(rawValue.Get())
-	if value < 0 {
-		value = 0
-	}
-
-	// Return 16-bit result from 12-bit value.
-	return uint16(value << 4)
+	return rawValue.Get()
```

Comment:
Can this break something? Like a client code out there in the wild expects 16bit value.

Reply:
Well, honestly this whole block makes no sense to me:

- Why convert a `Uint` to an `int` and then check if its `<0`? 
- Then `value << 4` does not convert something from 12 to 16 bit, it multiplies the value by $2^4 = 16$. Why?

I may not be aware of some special cases why this was done like so, maybe @aykevl (the original author) can shed some light on this.

Reply:
I believe the idea is (though I can't find this in documentation, probably picked up this from slack conversations) the value is always 16bit (i.e. ADC max is always 65535), but if hardware can't deliver that, so we simply have less resolution. This was client code does not need to worry about ADC MAX.

Reply:
Ahhh! Ohhh!... Ok, well then the shift must depend on the resolution. Can add that if you like. Then it remain compatible with current usages.

Reply:
Yes, correct, shift depends on ADC resolution.